### PR TITLE
replace golang/protobuf with  gogo/protobuf

### PR DESCRIPTION
**Pull request by krasi-georgiev**
fix for #3346

bumped all grpc-gateway packages to v1.2.2
updated and run  the `genproto.sh` script

the `grpc-gateway_out` is hard coded to use golang/protobuf so I updated the`genproto.sh` to replace the imports to gogo/protobuf/

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@brancz done

File: NOTICE [hunk 1]
```diff
@@ -52,7 +52,7 @@ Copyright jQuery Foundation and other contributors
 Licensed under the MIT License
 
 Go support for Protocol Buffers - Google's data interchange format
-http://github.com/golang/protobuf/
+http://github.com/gogo/protobuf/
```
Comment (beorn7) [inline NOTICE]: The gogo/protobuf has a different LICENSE text. It needs to be updated here.

Comment (krasi-georgiev) [reply NOTICE]: if you can please go ahead and makes these changes.
or let me know what should I change.

I don't even know what this NOTICE document is for :)

Comment (beorn7) [reply NOTICE]: It's a requirement of the Apache license.

Will update the license myself. Merging now...

Comment (beorn7) [reply NOTICE]: In different news, NOTICE is horribly out of date anyway. But at least we don't want plainly wrong copyright statements there. Already fixed in master.

**Pull request merged**